### ISSUE TITLE
[Bug]: Duplicate value from different relationship but seeing the same table

### ISSUE THREAD

**Issue by andrersilva96** [rappasoft/laravel-livewire-tables#1253]
### What happened?

Duplicate value from different relationship but seeing the same table.

### How to reproduce the bug

In a class/child table that has parent relationship, for example my class **Matcher**:

![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/1f2b57b4-5355-474e-8716-71562d9d93eb)

My datatable class has these columns:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/8276632f-ac35-41e0-9e18-c92069e2cff3)

The result:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/b1f0d54f-229d-49a6-8f58-4bdff85c67fd)

As you can see in database:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/c8aeafda-d370-461d-8321-636d87fb2f9e)

The properties table:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/57d5be02-ecf7-4335-b9b2-260dab62f376)

I won't to show the users table because is the same id from profiles table ok... so this is the profiles table:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/a79dc9c2-8aaa-4981-8f4a-f975023a3e0f)

The result should be:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/721afdfa-a99a-4e47-b5ef-97bbd916f5ea)

### Package Version

2.14.0

### PHP Version

8.1.x

### Laravel Version

9

### Alpine Version

3.10.5

### Theme

Bootstrap 5.x

### Notes

The result is the same for different relationship with same table, in the other words parent_id_from and parent_id_to for same table.

And Yes I've consulted your documentation:
https://rappasoft.com/docs/laravel-livewire-tables/v2/columns/relationships

### Error Message

None error.

**Comment by lrljoe** [comment]
Couple very quick question before I take a look in the next day or two - 

Are the Database:
- Column Data Type
- Column character_set
- Column collation
For all of these related fields exactly the same?

Are you using the Builder() approach or just a model name specified?

**Comment by andrersilva96** [comment]
Sorry take too long.

1. bigint unsigned null
2. utf8mb4
3. utf8mb4_0900_ai_ci
4. Yes
5. Yes, as you can see bellow:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/bee4f040-0ed2-4872-9d36-0c92dd6f1e3d)

**Comment by andrersilva96** [comment]
Sorry take too long.

1. bigint unsigned null
2. utf8mb4
3. utf8mb4_0900_ai_ci
4. Yes
5. Yes, as you can see bellow:
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/bee4f040-0ed2-4872-9d36-0c92dd6f1e3d)
6. I didn't declare nothing: 
![image](https://github.com/rappasoft/laravel-livewire-tables/assets/17456854/6ea59ac6-8e8e-4536-98d4-95550894235e)

And this is the query generated:
```
SELECT
  `matches`.`id` AS `id`,
  `properties`.`id` AS `propertyFrom.id`,
  `properties`.`id` AS `propertyTo.id`,
  `matches`.`created_at` AS `created_at`
FROM
  `matches`
  LEFT JOIN `properties` ON `matches`.`property_id_from` = `properties`.`id`
limit
  25 OFFSET 0;
```

**Comment by andrersilva96** [comment]
Searching deeper I realized that the problem is not in package, but deeper, in Laravel!
The fact is that laravel does not use aliases to create queries for example "FROM users AS u" so as I use 2 relationships from the same table with different ids it ends up falling into an in_array and ignores it if it has more than one like my case.

I came to a solution close, but nothing to resolve unfortunately. But if you have any tips, they are welcome!

**Comment by andrersilva96** [comment]
Solution:

To my column I called the label function and did something like:

```
->label(function ($row) use ($data) {
    $value = null;

    foreach (explode('.', $data['column']) as $attr) {
        $value = !$value ? $row->{$attr} : $value->{$attr};
    }

    return $value;
});
```

**Comment by lrljoe** [comment]
Glad you were able to fix it, but I'd definitely check your queries as you're possibly introducing multiple queries with your approach.

I'm a bit confused as to why you're checking if the model table is users in your Builder function, as you should surely know if it is or not when you create the Component...

You can use the following in your configure() method to add more fields that you're not directly referencing as columns, and then you'll be able to reference them as $row->fieldName, or whatever you choose.
```php
$this->setAdditionalSelects(['someTable.someColumnName as fieldName']);
```

**Comment by andrersilva96** [comment]
Thank you for helping, about the checking if the model table is users its because functionality of my system, instead of having several models of datatables I only have one and I pass the configuration with the model and the columns that I want in an array

**Comment by lrljoe** [comment]
Why not use a trait instead out of interest? 

 Otherwise if you're passing in things like columns as a variable, you run the risk of exposing something you don't want to to a malicious user.

**Comment by andrersilva96** [comment]
It because I use the nwidart modules, so each client it's a different configuration array and the user doesn't has access.

**Comment by lrljoe** [comment]
As long as they're protected/private variables in your component then!  If they're anything else, then the user will have access unless you're using another mechanism to protect the variable.

**Comment by lrljoe** [comment]
Just a quick note, I believe this is now fixed in v3.0, so I'd welcome your test when it comes to the beta in a week or two.

**Comment by lrljoe** [comment]
@andrersilva96  - The V3 Beta is out, if you're able to test it (it's Livewire 3.0+) then that'd be great, I can then look at backporting anything that isn't a breaking change into v2.
